README.md
# NAME

Dist::Zilla::MintingProfile::Iller::CSSON - Minting profile

<div>
    <p>
    <img src="https://img.shields.io/badge/perl-5.14+-blue.svg" alt="Requires Perl 5.14+" />
    <img src="https://img.shields.io/badge/coverage-100.0%25-brightgreen.svg" alt="coverage 100.0%" />
    <a href="https://github.com/Csson/p5-Dist-Zilla-MintingProfile-Iller-CSSON/actions?query=workflow%3Amakefile-test"><img src="https://img.shields.io/github/workflow/status/Csson/p5-Dist-Zilla-MintingProfile-Iller-CSSON/makefile-test" alt="Build status at Github" /></a>
    </p>
</div>

# VERSION

Version 0.0102, released 2020-12-28.

# SYNOPSIS

    dzil new -P Iller::CSSON New::Module

# DESCRIPTION

This mints a new distribution prepared for [Dist::Zilla](https://metacpan.org/pod/Dist::Zilla) via [Dist::Iller](https://metacpan.org/pod/Dist::Iller) and [Dist::Iller::Config::Author::CSSON](https://metacpan.org/pod/Dist::Iller::Config::Author::CSSON).

# SOURCE

[https://github.com/Csson/p5-Dist-Zilla-MintingProfile-Iller-CSSON](https://github.com/Csson/p5-Dist-Zilla-MintingProfile-Iller-CSSON)

# HOMEPAGE

[https://metacpan.org/release/Dist-Zilla-MintingProfile-Iller-CSSON](https://metacpan.org/release/Dist-Zilla-MintingProfile-Iller-CSSON)

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Dist-Zilla-MintingProfile-Iller-CSSON

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.0102 2020-12-28T12:40:20Z UTC
 [New Features]
 - Added a global doc to the template iller.yaml, where distribution_name
   will be set to the minted distribution's name. By also having a
   gitignore doc in iller.yaml, this will automatically create a .gitignore
   file where the minted distribution's build directories are excluded.

 [Stats & Dependency Changes]
 - 6 files changed, 84 insertions(+), 9 deletions(-)
 - (run req) ~ Carp (any) --> 1.38
 - (run req) ~ File::ShareDir (any) --> 1.102
 - (run req) ~ Moose 2.00 --> 2.1803
 - (run req) ~ namespace::autoclean (any) --> 0.22
 - (run req) ~ perl 5.010001 --> 5.014000
 - (test req) + IO::Handle (any)
 - (test req) + IPC::Open3 (any)
 - (dev req) + Badge::Depot 0.0103
 - (dev req) + Badge::Depot::Plugin::Coverage (any)
 - (dev req) + Badge::Depot::Plugin::Githubactions (any)
 - (dev req) + Badge::Depot::Plugin::Perl (any)
 - (dev req) + Dist::Zilla 6.015
 - (dev req) +
   Dist::Zilla::Plugin::Author::CSSON::GithubActions::Workflow::TestWithMakefile
   0.0106
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.009
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0200
 - (dev req) + Dist::Zilla::Plugin::Git 2.046
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.014
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::MetaConfig (any)
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::PromptIfStale 0.057
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Dist::Zilla::Plugin::Test::Compile 2.058
 - (dev req) + Dist::Zilla::Plugin::Test::EOF 0.0501
 - (dev req) + Dist::Zilla::Plugin::Test::EOL 0.18
 - (dev req) + Dist::Zilla::Plugin::Test::NoTabs 0.15
 - (dev req) + Dist::Zilla::Plugin::Test::Version 1.09
 - (dev req) + Pod::Weaver::Section::GenerateSection 1.01
 - (dev req) + Test::EOF (any)
 - (dev req) + Test::EOL (any)
 - (dev req) + Test::More 0.88
 - (dev req) + Test::NoTabs 1.4
 - (dev req) + Test::Version 1
 - (dev req) ~ Dist::Zilla::Plugin::CheckChangesHasContent (any) --> 0.011
 - (dev req) ~ Dist::Zilla::Plugin::Clean (any) --> 0.07
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild (any) --> 0.170880
 - (dev req) ~ Dist::Zilla::Plugin::Git::Contributors (any) --> 0.035
 - (dev req) ~ Dist::Zilla::Plugin::GithubMeta (any) --> 0.54
 - (dev req) ~ Dist::Zilla::Plugin::InstallRelease (any) --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::MetaProvides::Package (any) -->
   2.004003
 - (dev req) ~ Dist::Zilla::Plugin::PodWeaver (any) --> 4.006
 - (dev req) ~ Dist::Zilla::Plugin::ReadmeAnyFromPod (any) --> 0.163250
 - (dev req) ~ Dist::Zilla::Plugin::RunExtraTests (any) --> 0.029
 - (dev req) ~ Dist::Zilla::Plugin::Test::ReportPrereqs (any) --> 0.027
 - (dev req) ~ Pod::Weaver::Section::Badges (any) --> 0.0401
 - (dev req) ~ Test::Warnings (any) --> 0.026
 - (dev req) - Dist::Zilla::Plugin::CheckExtraTests
 - (dev req) - Dist::Zilla::Plugin::MetaProvides::Class
 - (dev req) - Dist::Zilla::Plugin::NextRelease
 - (dev req) - Dist::Zilla::Plugin::NextVersion::Semantic
 - (dev req) - Dist::Zilla::Plugin::OurPkgVersion
 - (dev req) - Dist::Zilla::Plugin::PodnameFromClassname
 - (dev req) - Dist::Zilla::Plugin::Prereqs::Plugins
 - (dev req) - Dist::Zilla::Plugin::PreviousVersion::Changelog
 - (dev req) - Dist::Zilla::Plugin::ReversionOnRelease
 - (dev req) - Dist::Zilla::Plugin::TravisYML
 - (dev req) - Pod::Weaver::Section::Homepage::DefaultCPAN
 - (dev req) - Pod::Weaver::Section::Source::DefaultGitHub
 - (dev sug) + Dist::Iller 0.1409
 - (dev sug) + Dist::Iller::Config::Author::CSSON 0.0328

0.0101 2016-01-20T20:46:58Z UTC
 [Bug Fixes]
 - Removed faulty Build.PL (fixes #1).

 [Enhancements]
 - Added simple test template.

 [Requirements]
 - Added missing prereqs: Carp, Dist::Zilla::Role::MintingProfile

0.0100 2016-01-14T23:40:08Z UTC
 - Initial release


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.031.
use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "CSSON's Dist::Zilla minting profile for Dist::Iller",
  "dist_author" => [
    "Erik Carlsson <info\@code301.com>"
  ],
  "dist_name" => "Dist-Zilla-MintingProfile-Iller-CSSON",
  "dist_version" => "0.0001",
  "license" => "perl",
  "module_name" => "Dist::Zilla::MintingProfile::Iller::CSSON",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "perl" => "5.010001"
  },
  "script_files" => [],
  "share_dir" => {
    "dist" => "profiles"
  },
  "test_requires" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => "0.96"
  }
);


my %fallback_build_requires = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Build" => "0.3601",
  "Test::More" => "0.96"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.017.
use strict;
use warnings;

use 5.014000;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "profiles";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Minting profile",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "Dist-Zilla-MintingProfile-Iller-CSSON",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.014000",
  "NAME" => "Dist::Zilla::MintingProfile::Iller::CSSON",
  "PREREQ_PM" => {
    "Carp" => "1.38",
    "Dist::Zilla::Role::MintingProfile" => "5.000",
    "File::ShareDir" => "1.102",
    "Moose" => "2.1803",
    "Path::Class" => 0,
    "namespace::autoclean" => "0.22"
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.0103",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => "1.38",
  "Dist::Zilla::Role::MintingProfile" => "5.000",
  "ExtUtils::MakeMaker" => 0,
  "File::ShareDir" => "1.102",
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => "2.1803",
  "Path::Class" => 0,
  "Test::More" => "0.96",
  "namespace::autoclean" => "0.22"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}
